### ISSUE TITLE
How to get updated word embedding by your model after feeding initialized word embedding into LSTM net

### ISSUE THREAD

**Issue by dodocho** [Linusp/soph#2]
I am relatively new to LSTM in keras, which I intend to apple to sequence labeling issue. Here is a little bit format description:

say, we have lots of samples (word sequences) and one of them is like "do you like the series game of thrones", and each word has a label (0/1) like this: 

do(0) you(1) like(1) the(0) series(1) game(1) of(0) thrones(1)

then, I use a BiLSTM to predict the labels given inputs, and the code may like this as follows:

**input_layer = Input(shape=(maxlen, ), dtype='int32')**

_#take pre-trained word embedding, "embedding_weights", as the initialized values_
**embedded = Embedding(input_dim = max_features+1, 
                                        output_dim = embedding_dim,
                                        weights = [embedding_weights],
                                        input_length = maxlen,
                                        mask_zero=True)(input_layer)**

_#build Bi-directional LSTM_
**forwards = LSTM(128, return_sequence=True)(embedded)
afterwards= LSTM(128, return_sequence=True, go_backwards=True)(embedded)
merged = merge([forwards, afterwards], mode='concat')
dense=TimeDistributed(Dense(128, activation='tanh'))(merged)
output_layer = TimeDistributed(Dense(2, activation='softmax'))(dense)
LSTM_model = Model(input=input_layer, output=output_layer)**

**LSTM_model.complie(optimizer='adam', loss='mse', metrics=['accuracy'])
LSTM_model.fit(train_data, train_label, batch_size=8, nb_epoch=10)**

I have 5,000 samples (pairs of word sequence and labels) around to train  model, and my questions is:

(1). Since I feed the initialized word embedding to the neural nets , it is expected that after training process, the word embedding in embedding layer would be updated by LSTM neural nets. but how could I output the the updated word embedding by neural nets? (weights parameter in embedding layer is actually a word vector matrix)

(2). I only have 5,000 samples (pairs of word sequence and label) and the length of each sample (word sequence) is about 10. In this case, would my model be too complicated for such small dataset? Also, I think there are a lot of open points above, so any suggestions would be greatly appreciated!

Again, thanks.

**Comment by jimmyangc** [comment]
@Linusp
